### PR TITLE
Ports: Add openttd

### PR DESCRIPTION
**Pull request by gunnarbeutner**
This adds a port definition for OpenTTD:

https://www.youtube.com/watch?v=8GzECfHjVtU :partying_face:

### CODE REVIEW

**Comment by gunnarbeutner** [comment]
https://www.youtube.com/watch?v=8GzECfHjVtU :partying_face:

**Comment by gunnarbeutner** [comment]
Before merging this I may have to remove the second commit depending on what happens with #6544.

**Comment by gunnarbeutner** [comment]
Alright, with #6544 merged this should be good to go. Rebased again to fix the conflict. :partying_face:

File: Ports/openttd-opengfx/package.sh [hunk 1]
```diff
@@ -0,0 +1,15 @@
+#!/usr/bin/env -S bash ../.port_include.sh
+port=openttd-opengfx
+version=0.6.1
+workdir=.
+files="https://cdn.openttd.org/opengfx-releases/${version}/opengfx-${version}-all.zip opengfx-${version}-all.zip c694a112cd508d9c8fdad1b92bde05e7c48b14d66bad0c3999e443367437e37e"
+auth_type=sha256
+
+build() {
+    tar xf opengfx-${version}.tar
```
Comment (linusg) [inline Ports/openttd-opengfx/package.sh:9]: wait why is this necessary? `files` are extracted automatically.

File: Ports/openttd-opengfx/package.sh [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+}
+
+install() {
+    mkdir -p ${SERENITY_INSTALL_ROOT}/usr/local/share/games/openttd/baseset/
+    cp -a opengfx-${version}/* ${SERENITY_INSTALL_ROOT}/usr/local/share/games/openttd/baseset/
```
Comment (linusg) [inline Ports/openttd-opengfx/package.sh:14]:
```suggestion
    run_nocd mkdir -p ${SERENITY_INSTALL_ROOT}/usr/local/share/games/openttd/baseset/
    run_nocd cp -a opengfx-${version}/* ${SERENITY_INSTALL_ROOT}/usr/local/share/games/openttd/baseset/
```

Let's get ourselves some logging :)

File: Ports/openttd-opensfx/package.sh [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+auth_type=sha256
+
+build() {
+    tar xf opensfx-${version}.tar
+}
+
+install() {
+    mkdir -p ${SERENITY_INSTALL_ROOT}/usr/local/share/games/openttd/baseset/
+    cp -a opensfx-${version}/* ${SERENITY_INSTALL_ROOT}/usr/local/share/games/openttd/baseset/
+}
```
Comment (linusg) [inline Ports/openttd-opensfx/package.sh:15]: whatever happens with the above comments, same here.

File: Ports/openttd-opengfx/package.sh (hunk 1 above)
Comment (gunnarbeutner) [reply Ports/openttd-opengfx/package.sh:9]: Well, they have a `.tar` archive in a `.zip` file and our script only handles the outer `.zip` file. :man_facepalming:

File: Ports/openttd-opengfx/package.sh (hunk 2 above)
Comment (gunnarbeutner) [reply Ports/openttd-opengfx/package.sh]: Done and done. :)

**Comment by linusg** [comment]
Awesome!

**Comment by jagger27** [comment]
If anyone has an issue running this port (like I did), make sure you increase the memory available to the VM. 

`export SERENITY_RAM_SIZE=2048M` did the trick for me.
